contracts/StablePriceOracle.sol

```diff
@@ -9,64 +9,75 @@ interface DSValue {
     function read() external view returns (bytes32);
 }
 
+
 // StablePriceOracle sets a price in USD, based on an oracle.
 contract StablePriceOracle is Ownable, PriceOracle {
     using SafeMath for *;
     using StringUtils for *;
 
-    // Oracle address
-    DSValue usdOracle;
-
-    // Rent in attodollars (1e-18) per second
+    // Rent in base price units by length. Element 0 is for 1-length names, and so on.
     uint[] public rentPrices;
 
+    // Oracle address
+    DSValue public usdOracle;
+
     event OracleChanged(address oracle);
+
     event RentPriceChanged(uint[] prices);
 
     constructor(DSValue _usdOracle, uint[] memory _rentPrices) public {
-        setOracle(_usdOracle);
+        usdOracle = _usdOracle;
         setPrices(_rentPrices);
     }
 
-    /**
-     * @dev Sets the price oracle address
-     * @param _usdOracle The address of the price oracle to use.
-     */
-    function setOracle(DSValue _usdOracle) public onlyOwner {
-        usdOracle = _usdOracle;
-        emit OracleChanged(address(_usdOracle));
+    function price(string calldata name, uint expires, uint duration) external view returns(uint) {
+        uint len = name.strlen();
+        if(len > rentPrices.length) {
+            len = rentPrices.length;
+        }
+        require(len > 0);
```

Comment:
This line can come above `(len > rentPrices.length) {`

Reply:
Can you also add message into `require` which helps debugging a lot

Reply:
I'm very much not a fan of this. It bloats contract size, and good tools ought to report the line the transaction failed on anyway.

Reply:
This is a bit out of scope of this PR but you may want to have a coding standard guideline because message on require has been used on `ShortNameClaims.sol`.

Other inconsistent coding conventions include the followings.

- Default indentation
- Mixture of uint/uint25
- Comment out of unused variable which is part of function signature (eg: `expire`).